object-store/src/main/java/org/hypertrace/config/objectstore/DefaultObjectStore.java

```diff
@@ -44,11 +45,11 @@ protected DefaultObjectStore(
     this.configChangeEventGeneratorOptional = Optional.empty();
   }
 
-  protected abstract Optional<T> buildObjectFromValue(Value value);
+  protected abstract Optional<T> buildDataFromValue(Value value);
 
-  protected abstract Value buildValueFromObject(T object);
+  protected abstract Value buildValueFromData(T data);
 
-  public Optional<T> getObject(RequestContext context) {
+  public Optional<T> getData(RequestContext context) {
```

Comment:
Not without introducing new grpc APIs - the original design of config service expected much more hierarchical config usage, so get (unlike upsert/delete/getAll) takes in any number of contexts and returns a merged result. That merged result has no associated context or metadata, since it may be a merge of multiple entries.

If we do have the need for timestamp on the singular get, we can introduce a getByContext rpc or something similar and then update this.

Reply:
Got it. I have never understood the merge usecase. However I agree, we can add another get call to return the data with timestamps. Can we add that call now or if need arises?

Reply:
I'd wait until the need arises, since it's purely additive to what we have here.

The merge use case is, for example - imagine fetching a config for an api where we may have some tenant-wide config, some service level config, and some api level config. We'd want to return a merged result conceptually like:
```java
Config getConfigForApi() {
  return apiConfig.withDefaultValuesFrom(serviceConfig.withDefaultValuesFrom(tenantConfig));
}
```